src/Teleportation.tsx

```diff
@@ -0,0 +1,166 @@
+        </mesh>
+      )}
+      <Interactive
+        onMove={(e) => {
+          if (
+            (e.target.inputSource.handedness === 'left' && !props.leftHand) ||
+            (e.target.inputSource.handedness === 'right' && !props.rightHand)
+          ) {
+            return
+          }
+
+          if (e.intersection) {
+            const distanceFromCamera = e.intersection.point.distanceTo(camera.position)
+            if (distanceFromCamera > maxDistanceTeleport) {
+              setSize(0)
+            } else {
+              setSize(MARKER_SIZE)
+            }
+            setIntersection(new Vector3(e.intersection.point.x, e.intersection.point.y, e.intersection.point.z))
+          }
+        }}
+        onHover={(e) => {
+          if (
+            (e.target.inputSource.handedness === 'left' && !props.leftHand) ||
+            (e.target.inputSource.handedness === 'right' && !props.rightHand)
+          ) {
+            return
+          }
+          if (e.intersection) {
+            const distanceFromCamera = e.intersection.point.distanceTo(camera.position)
+            if (distanceFromCamera > maxDistanceTeleport) {
+              setSize(0)
+            } else {
+              setSize(MARKER_SIZE)
+            }
+            setSize(MARKER_SIZE)
+          }
+        }}
+        onBlur={(e) => {
+          if (
+            (e.target.inputSource.handedness === 'left' && !props.leftHand) ||
+            (e.target.inputSource.handedness === 'right' && !props.rightHand)
+          ) {
+            return
+          }
+          setSize(0)
+        }}
+        onSelectStart={(e) => {
+          if (
+            (e.target.inputSource.handedness === 'left' && !props.leftHand) ||
+            (e.target.inputSource.handedness === 'right' && !props.rightHand)
+          ) {
+            return
+          }
+          if (e.intersection) {
+            const distanceFromCamera = e.intersection.point.distanceTo(camera.position)
+            if (distanceFromCamera > maxDistanceTeleport) {
+              setSize(0)
+            } else {
+              setSize(MARKER_SIZE * 1.1)
+            }
+          }
+        }}
+        onSelectEnd={(e) => {
+          setSize(MARKER_SIZE)
+          if (
+            (e.target.inputSource.handedness === 'left' && !props.leftHand) ||
+            (e.target.inputSource.handedness === 'right' && !props.rightHand)
+          ) {
+            return
+          }
+          if (intersection) {
+            const distanceFromCamera = intersection.distanceTo(camera.position)
+            if (distanceFromCamera <= maxDistanceTeleport) {
+              teleportTo(intersection)
+            }
+          }
+        }}
+      >
+        <mesh rotation={[-Math.PI / 2, 0, 0]} scale={[1000, 1000, 1000]}>
+          <planeBufferGeometry attach="geometry" args={[1, 1]} />
+          <meshBasicMaterial attach="material" transparent opacity={0} />
+        </mesh>
```

Comment:
We'll want to use `<planeGeometry />` as the *BufferGeometry nomenclature was removed in r144 and will be removed in r154 of three.js.